ivadomed/utils.py

```diff
@@ -236,14 +241,42 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[-1]):
+            data_im_tmp, _ = label(data_im[..., i], structure=bin_struct)
+            data_im_l.append(data_im_tmp)
         data_l_lst.append(data_im_l)
         del nib_im
+    data_l_lst = np.array(data_l_lst)
+    # channel went first due to the 'append' function
+    # Befor transpose we are supposed to have MC_simulation X Channel X Height X Width X Depth
+    # After transpose we expect to have MC_simulation X Height X Width X Depth X Channel
+    # this is wanted because the network output has a shape of  Channel X Height X Width X Depth which is saved as
```

Comment:
```suggestion
    # The rationale for doing this, is that the network output has the shape: Channel X Height X Width X Depth which is saved as
```

Reply:
after reading this explanation, i still don't understand the context for this chunk of code: do you want data_l_lst shape to match the output shape of the network? if so, how is "MC_simulation" the same as "Channel"?
Also, why talking about the nifti output afterwards-- is it relevant here?

Reply:
It is confusing indeed, the fact is I want them both to have a similar shape (like HxWxDxC) and we will loop over the MC simulation (so only the last 4 dimensions matters). We use HxWxDxC because that is the format of the input images (nifti_masks), we therefore have an output the same shape as the input (data ready to be save in nifti). 

I mentionned the network because of a mistake on the workflow of the function. I'll remove it.

Reply:
is what is written in 5efc0fe clearer ?

Reply:
nope-- comment added in the commit